### PR TITLE
Improvements to mock MutableData implementation

### PR DESCRIPTION
**Pull request by lionel1704**
- Improve mock impl. for MData by introducing a local enum in the mock vault impl.
- Add MsgIdConverter trait
- Add functions to the `Client` trait for other mutable data operations
with relevant tests

### CODE REVIEW

**Comment by lionel1704** [comment]
Closing and reopening to trigger Travis CI

File: safe_core/src/client/mock/vault.rs [hunk 1]
```diff
@@ -130,37 +132,38 @@ impl Vault {
         }
     }
 
-    pub fn authorised_read(
+    pub fn verify_requester(
```
Comment (nbaksalyar) [inline safe_core/src/client/mock/vault.rs]: Better to have this function in `safe-nd`, but it's OK to leave it here for now.

File: safe_core/src/client/mock/vault.rs [hunk 2]
```diff
@@ -231,32 +234,205 @@ impl Vault {
         dest: Authority<XorName>,
         payload: Vec<u8>,
     ) -> Result<(Authority<XorName>, Vec<u8>), ClientError> {
-        let request = unwrap!(deserialise(&payload));
+        let request: Request = unwrap!(deserialise(&payload));
+        dbg!(request.clone());
```
Comment (nbaksalyar) [inline safe_core/src/client/mock/vault.rs]: I think we don't need that? :slightly_smiling_face:

File: safe_core/src/client/mod.rs [hunk 3]
```diff
@@ -325,6 +383,120 @@ pub trait Client: Clone + 'static {
...
+            )
+        })
+        .and_then(|event| {
+            let res = match event {
+                CoreEvent::RpcResponse(res) => res,
+                _ => Err(CoreError::ReceivedUnexpectedEvent),
+            };
+            let result_buffer = unwrap!(res);
+            let res: Response<ClientError> = unwrap!(deserialise(&result_buffer));
+            dbg!(res.clone());
```
Comment (nbaksalyar) [inline safe_core/src/client/mod.rs]: Another one `dbg!` we need to remove (alternatively, convert it into `trace!` log call)

File: safe_core/src/client/mod.rs [hunk 4]
```diff
@@ -351,6 +523,90 @@ pub trait Client: Clone + 'static {
...
```
Comment (nbaksalyar) [inline safe_core/src/client/mod.rs]: One more. And there are some more after this one - could you pls also remove them?

**Pull request merged**